jolt-core/src/poly/commitment/dory.rs

```diff
@@ -0,0 +1,993 @@
+    transcript: Option<&'a mut T>,
+    _phantom: PhantomData<F>,
+}
+
+impl<'a, F: JoltField, T: Transcript> JoltToDoryTranscriptRef<'a, F, T> {
+    pub fn new(transcript: &'a mut T) -> Self {
+        JoltToDoryTranscriptRef {
+            transcript: Some(transcript),
+            _phantom: PhantomData,
+        }
+    }
+}
+
+impl<'a, F: JoltField, T: Transcript> DoryTranscript for JoltToDoryTranscriptRef<'a, F, T> {
+    type Scalar = JoltFieldWrapper<F>;
+
+    fn append_bytes(&mut self, label: &[u8], bytes: &[u8]) {
+        let transcript = self
+            .transcript
+            .as_mut()
+            .expect("Transcript not initialized");
+        let label_str: &'static [u8] = Box::leak(label.to_vec().into_boxed_slice());
+        transcript.append_message(label_str);
+        transcript.append_bytes(bytes);
+    }
+
+    fn append_field(&mut self, label: &[u8], x: &Self::Scalar) {
+        let transcript = self
+            .transcript
+            .as_mut()
+            .expect("Transcript not initialized");
+        let label_str: &'static [u8] = Box::leak(label.to_vec().into_boxed_slice());
+        transcript.append_message(label_str);
+        transcript.append_scalar(&x.0);
+    }
+
+    fn append_group<G: CanonicalSerialize>(&mut self, label: &[u8], g: &G) {
+        let transcript = self
+            .transcript
+            .as_mut()
+            .expect("Transcript not initialized");
+        let label_str: &'static [u8] = Box::leak(label.to_vec().into_boxed_slice());
+        transcript.append_message(label_str);
+        transcript.append_serializable(g);
+    }
+
+    fn append_serde<S: serde::Serialize>(&mut self, label: &[u8], s: &S) {
+        let transcript = self
+            .transcript
+            .as_mut()
+            .expect("Transcript not initialized");
+        let label_str: &'static [u8] = Box::leak(label.to_vec().into_boxed_slice());
+        transcript.append_message(label_str);
+        let bytes = postcard::to_allocvec(s).unwrap_or_default();
+        transcript.append_bytes(&bytes);
+    }
+
+    fn challenge_scalar(&mut self, label: &[u8]) -> Self::Scalar {
+        let transcript = self
+            .transcript
+            .as_mut()
+            .expect("Transcript not initialized");
+        let label_str: &'static [u8] = Box::leak(label.to_vec().into_boxed_slice());
+        transcript.append_message(label_str);
+        JoltFieldWrapper(transcript.challenge_scalar::<F>())
+    }
+
+    fn reset(&mut self, _domain_label: &[u8]) {
+        panic!("Reset not supported for JoltToDoryTranscript")
+    }
+}
+
+// ===== Type Aliases =====
+
+pub type JoltG1Wrapper = JoltGroupWrapper<G1Projective>;
+pub type JoltG2Wrapper = JoltGroupWrapper<G2Projective>;
+pub type JoltGTBn254 = JoltGTWrapper<Bn254>;
+
+pub type JoltG1MSM = JoltCustomMSM<G1Projective>;
+pub type JoltG2MSM = JoltCustomMSM<G2Projective>;
+pub type JoltGTMSM = JoltGTCustomMSM<Bn254>;
```

Comment:
You don't actually need these type aliases.
You can just have `pub struct JoltMSM;` instead of these `JoltCustomMSM<...>` and `JoltGTCustomMSM`, and write the implementations based on the generics:
```rust
impl<G> DoryMultiScalarMul<JoltGroupWrapper<G>> for JoltMSM
```
```rust
impl<P> DoryMultiScalarMul<JoltGTWrapper<P>> for JoltMSM
```
And rust will pick everything up automatically.